gui/IrodsExampleTab.py

```diff
@@ -7,58 +9,95 @@
 import PyQt6.uic
 
 import gui
-import meta
+import irodsConnector
 import utils
 from gui.irodsTreeView import IrodsModel
 
+
 class IrodsExampleTab(PyQt6.QtWidgets.QWidget,
-                   gui.ui_files.ExampleTab.Ui_ExampleTab):
+                      gui.ui_files.ExampleTab.Ui_ExampleTab):
+    """Example tab class.
+
+    """
+    conn = irodsConnector.manager.IrodsConnector()
     context = utils.context.Context()
+
     def __init__(self):
+        """
+
+        """
         super().__init__()
         if getattr(sys, 'frozen', False):
             super().setupUi(self)
         else:
             PyQt6.uic.loadUi("gui/ui_files/ExampleTab.ui", self)
+        self.ienv = self.context.irods_environment.config
```

Comment:
Do you agree that is better to refer to the ienv in this files with:  `self.context.irods_environment.config` to avoid local copies which could give problems?

Reply:
This is a single, static copy for the life of the class.  I initially chose to use a property as in `irodsConnector.session.Session`, but thought this might be seen as overkill.  The advantages of a property is that it reads the `config` fresh from the `Context` member each time, can be used for runtime documentation, and best of all, _made immutable_.

I also initially made the addition of these particular properties very simple by implementing an abstract base class that could be inherited from.  This, like too many _magic_ properties, proved undesirable as well.  That ABC was the `ContextContainer`.

I can re-implement those constructs if any of the advantages I mentioned are required.